src/test/kotlin/sirius/db/mongo/MongoFilterFactoryTest.kt

```diff
@@ -8,355 +8,543 @@
 
 package sirius.db.mongo
 
+import org.junit.jupiter.api.Test
+import org.junit.jupiter.api.assertThrows
+import org.junit.jupiter.api.extension.ExtendWith
 import sirius.db.mixing.Mapping
 import sirius.db.mongo.properties.MongoStringListEntity
-import sirius.kernel.BaseSpecification
+import sirius.kernel.SiriusExtension
 import sirius.kernel.commons.Value
 import sirius.kernel.di.std.Part
+import java.util.*
+import kotlin.test.assertEquals
+import kotlin.test.assertFalse
+import kotlin.test.assertTrue
 
-class MongoFilterFactorySpec extends BaseSpecification {
+@ExtendWith(SiriusExtension::class)
+class MongoFilterFactoryTest {
+    companion object {
+        @Part
+        private lateinit var mango: Mango
 
-    @Part
-    private static Mango mango
+        @Part
+        private lateinit var mongo: Mongo
 
-            @Part
-            private static Mongo mongo
+        private fun prefixSearch(query: String): Optional<Doc>? {
```

Comment:
Either use `Optional<Doc>` or `Doc?` here, using both Java and Kotlin optionals is a bit "doppelt gemoppelt". I would prefer `Doc?` as its more kotlin agnostic